assisted-events-scrape/workers/cluster_events_worker.py

```diff
@@ -49,14 +50,19 @@ def store_events_for_cluster(self, cluster: dict) -> None:
             if "hosts" not in cluster or len(cluster["hosts"]) == 0:
                 cluster["hosts"] = self.__get_hosts(cluster["id"])
             events = self.__get_events(cluster["id"])
-            component_versions = self._ai_client.get_versions()
+            component_versions = self.__get_versions()
             self._store_normalized_events(component_versions, cluster, events)
             self.cluster_events_storage.store(component_versions, cluster, events)
             self._config.changes.set_changed()
             log.debug(f'Storing events for cluster {cluster["id"]}')
         except Exception as e:
             self.__handle_unexpected_error(e, f'Error while processing cluster {cluster["id"]}')
 
+    @retry(ApiException, delay=1, tries=3, backoff=2, max_delay=4, jitter=1)
```

Comment:
Yes, it would. However this request is performed to `/v2/component_versions` - so if we had 404 there I guess there are bigger problems.
In the other two cases 404s are handled and exception risen again if it's another error (for now we could observe 503, 504 IIRC)

Reply:
In general I'd say we should only retry 5xx errors. 4xx is not just 404, but anything client did wrong (and I don't know of any known reason client might be flaky)

The reason I wouldn't want to retry on 4xx is to not pollute the logs and/or spend more time than needed. But if we don't have a separate exception for server-side errors then I guess we don't have much options here

Reply:
To summarize, not a blocker for the PR

Reply:
I totally agree with your point, in this case we get only one type of exception though.
I'll add some code to log non 5XX and raise exception in other cases, so that it can be retried.